### PR TITLE
fix input type, id, autocomplete

### PR DESCRIPTION
**Pull request by FlxMgdnz**
Adds correct input types and autocomplete properties

### CODE REVIEW

File: hanko-js/src/ui/pages/LoginPassword.tsx [hunk 1]
```diff
@@ -91,6 +91,8 @@ const LoginPassword = ({ userID, initialError }: Props) => {
           <InputText
             type={"password"}
             name={"password"}
+            autocomplete={"current-password"}
```
Comment (bjoern-m) [inline hanko-js/src/ui/pages/LoginPassword.tsx:94]: There is no autocompletion on my end. Tested with Safari and Chrome.

File: hanko-js/src/ui/pages/RegisterPassword.tsx [hunk 2]
```diff
@@ -76,6 +76,8 @@ const RegisterPassword = ({ user, registerAuthenticator }: Props) => {
         <InputText
           type={"password"}
           name={"password"}
+          autocomplete={"new-password"}
```
Comment (bjoern-m) [inline hanko-js/src/ui/pages/RegisterPassword.tsx:79]: There is no autocompletion on my end. Tested with Safari and Chrome.

**Review by bjoern-m — CHANGES_REQUESTED** [review]

File: hanko-js/src/ui/pages/LoginEmail.tsx [hunk 3]
```diff
@@ -164,6 +164,8 @@ const LoginEmail = () => {
         <InputText
           name={"email"}
           type={"email"}
+          autocomplete={"username"}
+          id={"email"}
```
[the remark this answers is not in the document]
Comment (FlxMgdnz) [reply hanko-js/src/ui/pages/LoginEmail.tsx:168]: I just applied tipps from https://web.dev/sign-in-form-best-practices/ to have maximum compatibility with password managers. I don't think the ids are actually making any difference, though, but I figured they don't do harm and may help some password manager to find the inputs.

File: hanko-js/src/ui/components/InputPasscodeDigit.tsx [hunk 4]
```diff
@@ -42,8 +42,8 @@ const InputPasscodeDigit = ({ index, focus, digit = "", ...props }: Props) => {
         // @ts-ignore
         part={"input passcode-input"}
         name={props.name + index.toString(10)}
-        autoComplete={"off"}
-        type={"text"}
+        autoComplete={"one-time-code"}
+        type={"numeric"}
```
[the remark this answers is not in the document]
Comment (FlxMgdnz) [reply hanko-js/src/ui/components/InputPasscodeDigit.tsx:46]: Yes I meant inputmode, my bad

File: hanko-js/src/ui/components/InputPasscodeDigit.tsx [hunk 5]
```diff
@@ -42,8 +42,8 @@ const InputPasscodeDigit = ({ index, focus, digit = "", ...props }: Props) => {
         // @ts-ignore
         part={"input passcode-input"}
         name={props.name + index.toString(10)}
-        autoComplete={"off"}
-        type={"text"}
+        autoComplete={"one-time-code"}
```
[the remark this answers is not in the document]
Comment (FlxMgdnz) [reply hanko-js/src/ui/components/InputPasscodeDigit.tsx:45]: With current shadow dom autocomplete limitations I fear we cannot test this at all. We probably won't need this at all as long as we don't add support for SMS OTP.

File: hanko-js/src/ui/pages/LoginEmail.tsx [hunk 6]
```diff
@@ -164,6 +164,8 @@ const LoginEmail = () => {
         <InputText
           name={"email"}
           type={"email"}
+          autocomplete={"username"}
```
[the remark this answers is not in the document]
Comment (FlxMgdnz) [reply hanko-js/src/ui/pages/LoginEmail.tsx:167]: That is does not change anything in Safari is most probably related to the shadow dom autocomplete limitations. From what I found, autocomplete=username should give the best compatibility with password managers in general.

File: hanko-js/src/ui/pages/LoginPassword.tsx (hunk 1 above)
Comment (FlxMgdnz) [reply hanko-js/src/ui/pages/LoginPassword.tsx:94]: It should be the correct autocomplete value for the password input, though. Unfortunately it does not help with password manager limitations with shadow dom.

File: hanko-js/src/ui/pages/RegisterPassword.tsx (hunk 2 above)
Comment (FlxMgdnz) [reply hanko-js/src/ui/pages/RegisterPassword.tsx:79]: Same as above

**Review by bjoern-m — APPROVED** [review]

**Pull request merged**